### PR TITLE
added fbeta score

### PR DESCRIPTION
**Pull request by subham-71**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py [hunk 1]
```diff
@@ -163,7 +170,7 @@ def get_classification_metrics(
         }
         return metrics
 
-    def get_metrics_classifier(
+    def _get_metrics_classifier(
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:173]: I didn't realise both are called within the same class. `_get_classification_metrics` and `_get_metrics_classifier` don't sound good. Can you rename one of these? I think we can rename `_get_metrics_classifier` to `_evaluate_classifier`. So:

1. `_get_classification_metrics`: This is a generic function on a given actual and predicted y
2. `_evaluate_classifier`: This is on the specific input model where we generate the predictions and are evaluating that classifier.

File: src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py [hunk 2]
```diff
@@ -87,9 +88,15 @@ def train_model(
                 "name": "roc_auc",
                 "function": roc_auc_score,
                 "greater_is_better": True,
-            }
+            },
+            {
+                "id": "fbeta_score",
+                "name": "fbeta_score",
+                "function": fbeta_score,
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: In the input config, we take the beta score param right, where is that being passed to the model here?

Comment (subham-71) [reply src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: The custom metrics are being passed to the model, and the metric will be added during setup. Also the metric_to_optimize will be used to sort the models.

Comment (dpatchigolla) [reply src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: How does the code read this [param](https://github.com/rudderlabs/rudderstack-profiles-classifier/blob/699f8036f45ca42c67b91e77687f26cffe9c6a65/src/predictions/rudderstack_predictions/config/model_configs.yaml#L21)?

Comment (subham-71) [reply src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: Not utilising this currently, Using the fbeta_score function from sklearn library.

Comment (subham-71) [reply src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: Do you want to utilize this parameter here [link](https://github.com/rudderlabs/rudderstack-profiles-classifier/blob/8ed3eeb08cdab1dfa6772f62890005ebfbe08d88/src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py#L155) ?

Comment (dpatchigolla) [reply src/predictions/rudderstack_predictions/trainers/ClassificationTrainer.py:95]: `fbeta_score` is the function, but this needs a param - the beta, which a user should be able to configure. Without it reading the param from the config, this is just equivalent to f1-score (with beta defaulting to 1)

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**